index.html

```diff
@@ -370,43 +378,87 @@
+        }
         let meta = data.data[entitytype][id];
         if (meta !== undefined) {
             text = text.replace(/‹cost›/, "Cost: " + cost(meta.Cost));
-            text = text.replace(/‹hp›/, "HP:&nbsp;" + meta.HP + ",");
-            text = text.replace(/‹attack›/, "Attack:&nbsp;" + meta.Attack + ",");
-            text = text.replace(/‹(A|a)rmor›/, "Armor:&nbsp;" + meta.MeleeArmor + ",");
-            text = text.replace(/‹(P|p)iercearmor›/, "Piercearmor:&nbsp;" + meta.PierceArmor + ",");
-            text = text.replace(/‹garrison›/, "Garrison:&nbsp;" + meta.GarrisonCapacity + ",");
-            text = text.replace(/‹range›/, "Range:&nbsp;" + meta.Range + ",");
-            text += minRangeIfDefined(meta.MinRange, "Min Range:&nbsp;");
-            text += ifDefined(meta.LineOfSight, "Line of Sight:&nbsp;");
-            text += ifDefined(meta.Speed, "Speed:&nbsp;");
-            text += secondsIfDefined(meta.TrainTime, "Build Time:&nbsp;");
-            text += secondsIfDefined(meta.ResearchTime, "Research Time:&nbsp;");
-            text += ifDefined(meta.FrameDelay, "Frame Delay:&nbsp;");
-            text += secondsIfDefined(meta.ReloadTime, "Reload Time:&nbsp;");
-            text += accuracyIfDefined(meta.AccuracyPercent, "Accuracy:&nbsp;");
-            text += arrayIfDefinedAndNonEmpty(meta.Attacks, "<br>Attacks:&nbsp;");
-            text += arrayIfDefinedAndNonEmpty(meta.Armours, "<br>Armours:&nbsp;");
+            let stats = []
+            if (text.match(/‹hp›/)) {
+                stats.push("HP:&nbsp;" + meta.HP);
+            }
+            if (text.match(/‹attack›/)) {
+                stats.push("Attack:&nbsp;" + meta.Attack);
+            }
+            if (text.match(/‹(A|a)rmor›/)) {
+                stats.push("Armor:&nbsp;" + meta.MeleeArmor);
+            }
+            if (text.match(/‹(P|p)iercearmor›/)) {
+                stats.push("Pierce armor:&nbsp;" + meta.PierceArmor);
+            }
+            if (text.match(/‹garrison›/)) {
+                stats.push("Garrison:&nbsp;" + meta.GarrisonCapacity);
+            }
+            if (text.match(/‹range›/)) {
+                stats.push("Range:&nbsp;" + meta.Range);
+            }
+            stats.push(minRangeIfDefined(meta.MinRange, "Min Range:&nbsp;"));
+            stats.push(ifDefined(meta.LineOfSight, "Line of Sight:&nbsp;"));
+            stats.push(ifDefined(meta.Speed, "Speed:&nbsp;"));
+            stats.push(secondsIfDefined(meta.TrainTime, "Build Time:&nbsp;"));
+            stats.push(secondsIfDefined(meta.ResearchTime, "Research Time:&nbsp;"));
+            stats.push(ifDefined(meta.FrameDelay, "Frame Delay:&nbsp;"));
+            stats.push(secondsIfDefined(meta.ReloadTime, "Reload Time:&nbsp;"));
+            stats.push(accuracyIfDefined(meta.AccuracyPercent, "Accuracy:&nbsp;"));
+            text = text.replace(/<p class="helptext__stats">(.+?)<\/p>/, "<h3>Stats</h3><p>" + stats.filter(Boolean).join(', ') + "<p>")
```

Comment:
This class in only used within this function; is it supposed to have some css styling associated to it?

Reply:
This helps identify the `<p>` for stats which I want to replace since there are multiple `<p>` in the text.